lib/packer/installer.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'open3'
+require 'digest/sha1'
+
+module Packer
+  class Installer
+    delegate :logger, :config, to: :@packer
+
+    def initialize(packer)
+      @packer = packer
+    end
+
+    def install
+      if stale?
+        record_package_digest
+        run_npm_install.tap do |success|
+          remove_package_digest unless success
+        end
+      else
+        true
+      end
+    end
+
+    def fresh?
+      package_file_digest == last_package_digest
+    end
+
+    def stale?
+      !fresh?
+    end
+
+    private
+
+    def last_package_digest
+      package_digest_path.read if package_digest_path.exist?
+    end
+
+    def package_file_digest
+      contents = if package_lock_path.exist?
```

Comment:
Imo a ternary operator is cleaner, but not a blocker

`contents = package_lock_path.exist? ? package_lock_path.read : ''`

Reply:
Fixed in https://github.com/simplybusiness/packer/pull/6/commits/4a13bf3f066b2e1741091dedfb37d44c8d6d6a44